### PR TITLE
[jvpvjp] Batch norm coverage with decomposition

### PR DESCRIPTION
**Pull request by samdow**
Adds decomposition from #675 for forward over reverse coverage. Similar to with layer norm, we needed to recompute the mean and variance so autograd propagates properly (sad) and needed to return tensors of zeros instead of None (sad)

### CODE REVIEW

**Review by zou3519 — APPROVED** [review]

File: functorch/_src/decompositions.py [hunk 1]
```diff
@@ -128,3 +134,84 @@ def native_layer_norm_backward(
...
+        grad_weight = dot_p * invstd
+    elif weight is not None:
+        grad_weight = torch.zeros_like(weight)  # should be None but doesn't work with vjp
+    else:
+        grad_weight = torch.zeros(())  # should be None but doesn't work with vjp
+
+    if output_mask[2]:
+        grad_bias = grad_output_sum
+    else:
+        grad_bias = torch.zeros_like(grad_output_sum)  # should be None but doesn't work with vjp
```
Comment (zou3519) [inline functorch/_src/decompositions.py:215]: sad, but is it what it is
